cli/src/semgrep/run_scan.py

```diff
@@ -300,6 +300,16 @@ def run_rules(
     )
 
 
+# This is used for testing and comparing with osemgrep.
+def list_targets_and_exit(target_manager: TargetManager, product: out.Product) -> None:
+    targets = target_manager.get_files_for_language(None, product)
+    for path in sorted(targets.kept):
+        print(f"+ {path}")
+    for (path, reason) in target_manager.ignore_log.list_skipped_paths_with_reason():
+        print(f"- [{reason}] {path}")
```

Comment:
maybe better to put the reason after the path

Reply:
I did that because I find the output a little easier to read this way, paths being of variable length. It looks like this:
```
- [semgrepignore_patterns_match] tests/tainting_rules/java/try_return.yaml
- [semgrepignore_patterns_match] tests/tainting_rules/js/await.js
- [semgrepignore_patterns_match] tests/tainting_rules/js/await.yaml
- [semgrepignore_patterns_match] tests/tainting_rules/js/call_chain.js
- [semgrepignore_patterns_match] tests/tainting_rules/js/call_chain.yaml
- [semgrepignore_patterns_match] tests/tainting_rules/js/eslint_obj_inj.js
- [semgrepignore_patterns_match] tests/tainting_rules/js/eslint_obj_inj.yaml
- [semgrepignore_patterns_match] tests/tainting_rules/js/everything_source.js
- [semgrepignore_patterns_match] tests/tainting_rules/js/everything_source.yaml
- [semgrepignore_patterns_match] tests/tainting_rules/js/everything_source1.js
- [semgrepignore_patterns_match] tests/tainting_rules/js/everything_source1.yaml
- [semgrepignore_patterns_match] tests/tainting_rules/js/metavar_eq_conditional.js
- [semgrepignore_patterns_match] tests/tainting_rules/js/metavar_eq_conditional.yaml
- [semgrepignore_patterns_match] tests/tainting_rules/js/metavar_eq_simple.js
- [semgrepignore_patterns_match] tests/tainting_rules/js/metavar_eq_simple.yaml
- [semgrepignore_patterns_match] tests/tainting_rules/js/nested_function.js
- [semgrepignore_patterns_match] tests/tainting_rules/js/nested_function.yaml
- [semgrepignore_patterns_match] tests/tainting_rules/js/sanitized_by_side_effect.js
- [semgrepignore_patterns_match] tests/tainting_rules/js/sanitized_by_side_effect.yaml
- [semgrepignore_patterns_match] tests/tainting_rules/js/simpl_nodejs_eval.js
- [semgrepignore_patterns_match] tests/tainting_rules/js/simpl_nodejs_eval.yaml
- [semgrepignore_patterns_match] tests/tainting_rules/js/single_variable_source.js
- [semgrepignore_patterns_match] tests/tainting_rules/js/single_variable_source.yaml
- [semgrepignore_patterns_match] tests/tainting_rules/js/throw.js
- [semgrepignore_patterns_match] tests/tainting_rules/js/throw.yaml
- [semgrepignore_patterns_match] tests/tainting_rules/php/break.php
- [semgrepignore_patterns_match] tests/tainting_rules/php/break.yaml
- [semgrepignore_patterns_match] tests/tainting_rules/php/echo.php
- [semgrepignore_patterns_match] tests/tainting_rules/php/echo.yaml
- [semgrepignore_patterns_match] tests/tainting_rules/php/lval_var_sink.php
- [semgrepignore_patterns_match] tests/tainting_rules/php/lval_var_sink.yaml
- [semgrepignore_patterns_match] tests/tainting_rules/php/no_duplicate_submatches.php
- [semgrepignore_patterns_match] tests/tainting_rules/php/no_duplicate_submatches.yaml
- [semgrepignore_patterns_match] tests/tainting_rules/php/switch.php
- [semgrepignore_patterns_match] tests/tainting_rules/php/switch.yaml
- [semgrepignore_patterns_match] tests/tainting_rules/php/ternary.php
- [semgrepignore_patterns_match] tests/tainting_rules/php/ternary.yaml
- [semgrepignore_patterns_match] tests/tainting_rules/python/break.py
- [semgrepignore_patterns_match] tests/tainting_rules/python/break.yaml
- [semgrepignore_patterns_match] tests/tainting_rules/python/fake_tokens.py
- [semgrepignore_patterns_match] tests/tainting_rules/python/fake_tokens.yaml
- [semgrepignore_patterns_match] tests/tainting_rules/python/raise_from.py
- [semgrepignore_patterns_match] tests/tainting_rules/python/raise_from.yaml
- [semgrepignore_patterns_match] tests/tainting_rules/python/raise_no_args.py
- [semgrepignore_patterns_match] tests/tainting_rules/python/raise_no_args.yaml
- [semgrepignore_patterns_match] tests/tainting_rules/python/simpl_django_redirect.py
- [semgrepignore_patterns_match] tests/tainting_rules/python/simpl_django_redirect.yaml
- [semgrepignore_patterns_match] tests/tainting_rules/python/sink_param.py
- [semgrepignore_patterns_match] tests/tainting_rules/python/sink_param.yaml
- [semgrepignore_patterns_match] tests/tainting_rules/python/sink_return.py
- [semgrepignore_patterns_match] tests/tainting_rules/python/sink_return.yaml
- [semgrepignore_patterns_match] tests/tainting_rules/python/source_param.py
- [semgrepignore_patterns_match] tests/tainting_rules/python/source_param.yaml
- [semgrepignore_patterns_match] tests/tainting_rules/python/tainting_top.py
- [semgrepignore_patterns_match] tests/tainting_rules/python/tainting_top.yaml
- [semgrepignore_patterns_match] tests/tainting_rules/python/try_finally.py
- [semgrepignore_patterns_match] tests/tainting_rules/python/try_finally.yaml
- [semgrepignore_patterns_match] tests/tainting_rules/python/try_finally1.py
- [semgrepignore_patterns_match] tests/tainting_rules/python/try_finally1.yaml
- [semgrepignore_patterns_match] tests/tainting_rules/python/try_raise.py
- [semgrepignore_patterns_match] tests/tainting_rules/python/try_raise.yaml
- [semgrepignore_patterns_match] tests/tainting_rules/ruby/switch.rb
- [semgrepignore_patterns_match] tests/tainting_rules/ruby/switch.yaml
- [semgrepignore_patterns_match] tests/tainting_rules/scala/new_object_sink.scala
- [semgrepignore_patterns_match] tests/tainting_rules/scala/new_object_sink.yaml
- [semgrepignore_patterns_match] tests/tainting_rules/ts/source_exp.ts
- [semgrepignore_patterns_match] tests/tainting_rules/ts/source_exp.yaml
- [semgrepignore_patterns_match] tests/typing/BasicParam.java
- [semgrepignore_patterns_match] tests/typing/ClassFields.java
- [semgrepignore_patterns_match] tests/typing/EqVarCmp.java
- [semgrepignore_patterns_match] tests/typing/FuncParam.go
- [semgrepignore_patterns_match] tests/typing/PropVarDef.go
- [semgrepignore_patterns_match] tests/typing/StaticVarDef.go
- [semgrepignore_patterns_match] tests/typing/VarDef.java
- [semgrepignore_patterns_match] tmp/foo
- [semgrepignore_patterns_match] tmp/foo.hack
- [semgrepignore_patterns_match] tmp/foo.php
- [semgrepignore_patterns_match] tmp/foo1.js
- [semgrepignore_patterns_match] tmp/foo2.js
- [semgrepignore_patterns_match] tmp/foo3.bar
- [semgrepignore_patterns_match] tmp/hello1.bash
- [semgrepignore_patterns_match] tmp/hello2
- [semgrepignore_patterns_match] tmp/hello3
- [semgrepignore_patterns_match] tmp/hello4
- [semgrepignore_patterns_match] tmp/hello5
- [semgrepignore_patterns_match] tmp/hello6
```

Reply:
I don't love this output format overall but we can change it anytime.